x/cellarfees/keeper/pool.go

```diff
@@ -0,0 +1,56 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/peggyjv/sommelier/v4/app/params"
+	"github.com/peggyjv/sommelier/v4/x/cellarfees/types"
+)
+
+// Appends the coin to the pool coins if denom isn't already present, otherwise add the amount to the
+// existing balance
+func (k Keeper) AddCoinToPool(ctx sdk.Context, coin sdk.Coin) {
+	// (Collin) should an empty coin error?
+	if (coin == sdk.Coin{} || coin.IsZero()) {
+		return
+	}
+
+	if coin.Denom == params.BaseCoinUnit {
+		panic("Cannot add SOMM to cellar fee pool")
```

Comment:
Perhaps it might be better to return an error here?

Reply:
I think an error would imply this is an expected case, but I added it as a sanity check since there should be no public API code paths that use `AddCoinToPool()`. If the protocol is attempting to add `usomm` to the fee pool, there is a programmatic error somewhere in the module and it seems to me that warrants a panic? It could maybe be argued we don't need the check but I tend to favor sanity checks for readability at least.